SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -263,7 +263,10 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 
 	ComponentDatas.Append(DynamicComponentDatas);
 
-	ComponentDatas.Add(NetDriver->InterestFactory->CreateInterestData(Actor, Info, EntityId));
+	bool bOwnerReady;
+	ComponentDatas.Add(NetDriver->InterestFactory->CreateInterestData(Actor, Info, EntityId, bOwnerReady));
```

Comment:
It seems fairly trivial to check `bOwnerReady` without requiring the bool to be passed all the way through this stack. Perhaps write a utility function that returns if it's ready? Or we could keep the existing interface, but add a `IsOwnerReady` function on the interest factory, and use that instead.

Reply:
It is, but in terms of balancing (minimal interface change/minimal additional work), it looked like a good tradeoff.
To do the check, I have to grab the owner, and get its entity id. Are you suggesting I do this work twice, once to setup interest, and a second time to check ?

Reply:
To keep the same amount of work, I could use  TOptional<WorkerEntityId> GetEntityForOwnerInterest(), and pass that to the CreateInterest function.